v2/pkg/cli/delete.go

```diff
@@ -0,0 +1,288 @@
+	flagRetryOpts, retryOpts := mirror.RetryFlags()
+
+	opts := &mirror.CopyOptions{
+		Global:              global,
+		DeprecatedTLSVerify: deprecatedTLSVerifyOpt,
+		SrcImage:            srcOpts,
+		DestImage:           destOpts,
+		RetryOpts:           retryOpts,
+		Function:            string(mirror.DeleteMode),
+	}
+
+	mkd := MakeDir{}
+	ex := &ExecutorSchema{
+		Log:     log,
+		Opts:    opts,
+		MakeDir: mkd,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Deletes all related images and manifests from a remote repository or local cache or both",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := ex.ValidateDelete()
+			if err != nil {
+				log.Error("%v ", err)
+				os.Exit(1)
+			}
+			err = ex.CompleteDelete()
+			if err != nil {
+				log.Error("%v ", err)
+				os.Exit(1)
+			}
+			// prepare internal storage
+			err = ex.setupLocalStorage()
+			if err != nil {
+				log.Error(" %v ", err)
+				os.Exit(1)
+			}
+
+			err = ex.RunDelete(cmd)
+			if err != nil {
+				log.Error("%v ", err)
+				os.Exit(1)
+			}
+		},
+	}
+	cmd.PersistentFlags().StringVarP(&opts.Global.ConfigPath, "config", "c", "", "Path to delete imageset configuration file")
+	cmd.PersistentFlags().StringVarP(&opts.Global.DeleteSource, "source", "s", "", "The working directory used to do the initial mirroring")
+	cmd.Flags().StringVar(&opts.Global.LogLevel, "loglevel", "info", "Log level one of (info, debug, trace, error)")
+	cmd.Flags().StringVar(&opts.Global.DeleteID, "delete-id", "", "Used to append to artifacts created by the delete functioality")
+	cmd.Flags().BoolVar(&opts.Global.SkipCacheDelete, "skip-cache-delete", ex.Opts.Global.SkipCacheDelete, "Used to skip deleting of local cache manifests and blobs")
+	cmd.Flags().BoolVar(&opts.Global.SkipRegistryDelete, "skip-registry-delete", ex.Opts.Global.SkipRegistryDelete, "Used to skip deleting of remote registry images")
+	cmd.Flags().StringVar(&opts.Global.DeleteDestination, "destination", "", "The remote registry to delete from (optional)")
+	cmd.Flags().StringVar(&opts.Global.WorkingDir, "dir", workingDir, "Assets directory")
+	cmd.Flags().Uint16VarP(&opts.Global.Port, "port", "p", 55000, "HTTP port used by oc-mirror's local storage instance")
+	cmd.Flags().BoolVar(&opts.Global.V2, "v2", ex.Opts.Global.V2, "Redirect the flow to oc-mirror v2 - PLEASE DON'T USE it. V2 is still under development and it is not production ready.")
+	cmd.Flags().BoolVar(&opts.Global.DryRun, "dry-run", ex.Opts.Global.DryRun, "If set will only output the list of images (manifests and shas) to delete")
+	// nolint: errcheck
+	cmd.Flags().MarkHidden("v2")
+	cmd.Flags().AddFlagSet(&flagSharedOpts)
+	cmd.Flags().AddFlagSet(&flagRetryOpts)
+	cmd.Flags().AddFlagSet(&flagDepTLS)
+	cmd.Flags().AddFlagSet(&flagSrcOpts)
+	cmd.Flags().AddFlagSet(&flagDestOpts)
+
+	// hide flags
+	HideFlags(cmd)
+
+	return cmd
+}
+
+// Validate - cobra validation
+func (o ExecutorSchema) ValidateDelete() error {
```

Comment:
I did not understand why it is needed to duplicate the cli code, is there anything I am missing maybe?

Reply:
Yes , I have introduced new flags and these combinations need to be validated and checked